src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IRequestContextFactory.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface IRequestContextFactory<TRequestContext>
+{
+    Task<TRequestContext?> CreateRequestContextAsync(IQueueItem<TRequestContext> queueItem, CancellationToken queueCancellationToken, CancellationToken requestCancellationToken);
```

Comment:
def doc these two tokens.  i'm also surprised we would ever expose thsi downwards (as opposed to makign a linked token internally that we pass downwards).

Reply:
+1 - the queue cancellation token is inherently an impl of the queue - someone who implements a request factory shouldn't need to care where cancellation came from (shutdown, or request cancellation)